v2/app/incubator/raspberrypi/basic_all.py

```diff
@@ -214,21 +214,25 @@ def main():
     sensor_time = (time.time() - SENSOR_FREQ_SECONDS)
     while True:
         adjust_lights()
-        camera_delta = (time.time() - camera_time)
-        sensor_delta = (time.time() - sensor_time)
         # take picture if necessary
-        if (camera_delta > CAMERA_FREQ_SECONDS):
-            camera_time += CAMERA_FREQ_SECONDS
+        camera_delta = (time.time() - camera_time)
+        if ((camera_delta > CAMERA_FREQ_SECONDS) or (camera_retry > 0 and camera_retry < max_retry)):
             if is_lights_on():
-                took_pic = take_picture()
-                if not took_pic:
-                    camera_time = (time.time() - CAMERA_FREQ_SECONDS)
+                if take_picture():
+                    camera_time, camera_retry = (time.time(), 0)
```

Comment:
I see why you are setting camera_time to current time because you don't know how many failures you might have had so you don't really know where you are in time relative to the last successful picture but this will result in additional time added to each interval equal to the picture upload time in normal operation.  We had this "time drift" before and I think we fixed it by incrementing camera_time before the picture upload.  So maybe the logic you had before where you increment camera_time before you take the picture and then reverse that change if the picture taking fails is needed?